src/linagora.esn.unifiedinbox/app/components/mailbox-shared-settings/mailbox-shared-settings.controller.js

```diff
@@ -54,19 +54,19 @@ function InboxMailboxSharedSettingsController(
       owner.displayName = userUtils.displayNameOf(owner);
       self.owner = owner;
     });
-    getUserSharedInformation(self.mailbox.sharedWith);
+    getUserSharedInformation(self.mailbox.rights);
   }
 
   function getOwner() {
-    var owner;
+    if (self.mailbox.namespace.toLowerCase().startsWith('delegated')) {
+      const ownerEmail = self.mailbox.namespace.replace(/^.*\[(.*)\]$/g, '$1');
```

Comment:
OMG. What is this magic regex please ?

Reply:
This is a magic regex to get owner from namespace. I put it in a function.